bridge-worker/store.go

```diff
@@ -0,0 +1,122 @@
+package bridgeworker
+
+import (
+	"fmt"
+
+	"cosmossdk.io/log"
+	"cosmossdk.io/math"
+	bridgetypes "github.com/mezo-org/mezod/x/bridge/types"
+	"github.com/supabase-community/supabase-go"
+)
+
+const (
+	AssetsUnlockedEventsTable  = "assets_unlocked_events"
+	AttestationSignaturesTable = "attestations_signatures"
+)
+
+type SupabaseStore struct {
+	logger log.Logger
+	client *supabase.Client
+}
+
+func NewSupabaseStore(
+	logger log.Logger,
+	databaseURL string,
+	anonKey string,
+) (*SupabaseStore, error) {
+	client, err := supabase.NewClient(databaseURL, anonKey, &supabase.ClientOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to create supabase client: %w", err)
+	}
+
+	return &SupabaseStore{
+		logger: logger,
+		client: client,
+	}, nil
+}
+
+// SaveAttestation will save an attestation in the database
+// because there's no good atomic way of checking the error that
+// occurred, we take an optimistic approach, meaning:
+// we first try to insert
```

Comment:
If we know saving duplicated data will error out, I think it's better to try load the data before save and skip in case the data already exist.